code/modules/paperwork/paper.dm

```diff
@@ -351,6 +234,57 @@
 	..()
 	cut_overlay("paper_onfire_overlay")
 
+/obj/item/paper/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		var/datum/asset/assets = get_asset_datum(/datum/asset/spritesheet/simple/paper)
+		assets.send(user)
+		/// The x size is because we double the width for the editor
+		ui = new(user, src, ui_key, "PaperSheet", name, 400, 600, master_ui, state)
+		ui.set_autoupdate(FALSE)
+		ui.open()
+
+/obj/item/paper/ui_close(mob/user)
+	var/datum/tgui/ui = SStgui.try_update_ui(user, src, "main");
+	if(ui)
+		ui.close()
+
+/obj/item/paper/proc/ui_update()
+	var/datum/tgui/ui = SStgui.try_update_ui(usr, src, "main");
+	if(ui)
+		ui.update()
+
+/obj/item/paper/ui_data(mob/user)
+	var/list/data = list()
+	data["text"] = info
+	data["paper_state"] = icon_state	/// TODO: show the sheet will bloodied or crinkling
+	data["pen_color"] = pen_color
+	data["paper_color"] = color || "white"	// color might not be set
+	data["edit_sheet"] = readonly || finalized ? FALSE : TRUE
+	/// data["stamps_info"] = list(stamp_info)
+	data["stamps"] = stamps
+	return data
+
+
+/obj/item/paper/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("save")
+			var/in_paper = params["text"]
+			if(length(in_paper) > 0 && length(in_paper) < 1000) // Sheet must have less than 1000 charaters
```

Comment:
you could just trim the length

Reply:
Ah well this is to detect if someone is using something to inject text into the html stream.  The javascript IS limiting the size sent to byond, so if its more than the 1000, then someone is doing something funky.  So its why I kind of want to log it and not store it.